addons/account/models/chart_template.py

```diff
@@ -1089,13 +1089,42 @@ def _get_account_reconcile_model(self, template_code):
     # Tooling
     # --------------------------------------------------------------------------------
 
-    def ref(self, xmlid, raise_if_not_found=True):
-        if '.' in xmlid:
-            return self.env.ref(xmlid, raise_if_not_found)
-        return (
-            self.env.ref(f"account.{self.env.company.id}_{xmlid}", raise_if_not_found=False)
-            or self.env.ref(f"account.{self.env.company.parent_ids[0].id}_{xmlid}", raise_if_not_found)
-        )
+    def ref(self, xml_id, raise_if_not_found=True):
+        return self.refs([xml_id], raise_if_not_found)
+
+    def refs(self, xml_ids, raise_if_not_found=True):
+        """
+            When giving xml_ids, three steps are tried:
+                1. if <module>.<record_id> is given, then we suppose the xml_ids are already correct
+                2. if <module> is not given, `account` is assumed, and the record is searched
+                   in the company's CoA by <module>.<company_id>_<record_id>
+                3. if not found, we use <module>.<parent_company_id>_<record_id>
+            You cannot mix these kinds of xml_ids input, it's either 1, 2 or 3.
+        """
+        company, parent_company = self.env.company, self.env.company.parent_ids[0]
+        parent_company = parent_company if parent_company != company else False
+
+        dots, non_dots = [], []
```

Comment:
You can use `partition` from odoo.tools.
Note that if you have a mix, you get refs only from `dots`.

Reply:
We could just fail!